syntax = "proto3";
package print_nanny.common;

message Window {
    float window_start = 1;
    float window_end = 2;
}

message OctoprintEnvironment {
    string client_version = 1;
    string plugin_version = 2;
    string python_version = 3;
    string pip_version = 4;
    string octopi_version = 5;
    string octoprint_version = 6;
    string virtualenv = 7;
    string platform = 8;
    uint32 bits = 9;
    uint32 cores = 10;
    float freq = 11;
    uint64 ram = 12;
    string pi_model = 13;
    string pi_throttle_state = 14;

}

message PrintSession {
    string session = 1;
    int32 id = 2;
    double created_ts = 3;
    string datesegment = 4;
}

message Metadata {
    double ts = 1;
    uint32 user_id = 2;
    uint32 octoprint_device_id = 3;
    uint64 cloudiot_device_id = 4;
    OctoprintEnvironment octoprint_environment = 5;
    PrintSession print_session = 6;
}